packages/desktopjs/src/Default/default.ts

```diff
@@ -326,13 +325,17 @@ export namespace Default {
             }
 
             const window = this.globalWindow.open(url, target, features);
-            window[Container.windowOptionsPropertyKey] = options; // Store the original options
+            const isSameOrigin = this.isSameOrigin(window);
+
+            if (isSameOrigin) {
+                window[Container.windowOptionsPropertyKey] = options; // Store the original options
 
-            // Set name as a unique desktopJS name property instead of overloading window.name
-            window[DefaultContainer.windowNamePropertyKey] = newOptions.name;
+                // Set name as a unique desktopJS name property instead of overloading window.name
+                window[DefaultContainer.windowNamePropertyKey] = newOptions.name;
+            }
 
             const newWindow = this.wrapWindow(window);
-            this.emit("window-created", { sender: this, name: "window-created", window: newWindow, windowId: newWindow.id, windowName: newOptions.name });
+            this.emit("window-created", { sender: this, name: "window-created", window: newWindow, windowId: isSameOrigin ? newWindow.id : undefined, windowName: newOptions.name });
```

Comment:
Codacy found an issue: [Unexpected use of undefined.](https://app.codacy.com/manual/brian.ingenito/desktopJS/pullRequest?prid=6294145)